### PR TITLE
[Merged by Bors] - chore(AlgebraicGeometry.Morphisms.RingHomProperties): speedups

### PR DESCRIPTION
**Pull request by collares**
I don't really understand this file, but I was bothered by the sole declaration requiring more than a million heartbeats. I noticed strategically adding a few `conv`s can make `sourceAffineLocally_of_source_open_cover_aux` not require a bump.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by collares** [comment]
!bench

**Comment by collares** [comment]
!bench

**Comment by semorrison** [comment]
bors merge

File: Mathlib/AlgebraicGeometry/Morphisms/RingHomProperties.lean [hunk 1]
```diff
@@ -265,7 +265,6 @@ theorem sourceAffineLocally_isLocal (h₁ : RingHom.RespectsIso @P)
 variable (hP : RingHom.PropertyIsLocal @P)
 
 -- Porting note: the terms here are getting huge ~ 1/2 Gb for the goal midway (with `pp.explicit`)
-set_option maxHeartbeats 4000000 in
 theorem sourceAffineLocally_of_source_open_cover_aux (h₁ : RingHom.RespectsIso @P)
     (h₃ : RingHom.OfLocalizationSpanTarget @P) {X Y : Scheme} (f : X ⟶ Y) (U : X.affineOpens)
     (s : Set (X.presheaf.obj (op U.1))) (hs : Ideal.span s = ⊤)
```
Comment (alreadydone) [inline Mathlib/AlgebraicGeometry/Morphisms/RingHomProperties.lean:270]: In #7747 by the original author @erdOne this was golfed (apparently using just one new lemma introduced).

Comment (collares) [reply Mathlib/AlgebraicGeometry/Morphisms/RingHomProperties.lean:270]: Ah, I wish I knew that beforehand :( I'd be happy to have my hacky proof replaced by a proper one, of course.

**Pull request closed without merging**